Add experimental options for more parallelization in webpack builds (#60177)
vercel/next.js: errors/parallel-build-without-worker.mdx

---
title: Parallel Build Without Build Worker
---

## Why This Error Occurred

> The `experimental.parallelServerAndEdgeCompiles` and `experimental.parallelServerBuildTraces`

Reviewer: Rename to `parallelServerCompiles`? "nodejs" and "edge" are runtime but are both "server" target

Author: Done.

> options require that the `experimental.webpackBuildWorker` option is set to true. These

Reviewer 2: Small nit 

```suggestion
options require that the `experimental.webpackBuildWorker` option is set to `true`. These
```

> options use workers to improve the paralellization of the build which may improve performance,

Reviewer 2: ```suggestion
options use workers to improve the parallelization of the build which may improve performance,
```

> The `experimental.parallelServerAndEdgeCompiles` and `experimental.parallelServerBuildTraces`
> options require that the `experimental.webpackBuildWorker` option is set to true. These
> options use workers to improve the paralellization of the build which may improve performance,
> but the build may use more memory at the same time.

Reviewer 2: Small formatting:

```suggestion
The `experimental.parallelServerAndEdgeCompiles` and `experimental.parallelServerBuildTraces` options require that the `experimental.webpackBuildWorker` option is set to `true`. These options use workers to improve the parallelization of the build, which may improve performance, but the build may use more memory at the same time.
```

## Possible Ways to Fix It

> Build workers are enabled by default unless you have a custom webpack config.
> You can force enable the option by setting `config.experimental.webpackBuildWorker: true` in the config,
> but some webpack configurations may not be compatible.

Reviewer 2: ```suggestion
Build workers are enabled by default unless you have a custom webpack config. You can force enable the option by setting `experimental.webpackBuildWorker: true` in your `next.config.js` file, but some webpack configuration options may not be compatible.
```
